### ISSUE TITLE
contact app 1.5.1 constructs/shows new groups

### ISSUE THREAD

**Issue by severalthings** [owncloud/contacts#602]
This error must have been introduced via an update. So the "steps to reproduce" won't work, but they explain the result.

### Steps to reproduce
1. create 3 contacts X,Y,Z
2. create 2 groups A and B
3. put X in A, Y in B, and Z in A and B

### Expected behaviour
On the left side of the web-interface there should be tw groups.
When accessing the groups / contacts with thunderbird/category manager2/ InverseSogo Connector the groups are shown correctly.

### Actual behaviour
There are three groups A, B and AB. Each group has one member. I.E. any combinations of groups that a contact belongs to leads to a group shown in the web-interface

### Server configuration
**Operating system**: Ubuntu 16.04

**Web server:**
apache
**Database:**

**PHP version:**

**ownCloud version:** (see ownCloud admin page)
9.1.6
**Contacts version:** (see ownCloud apps page)
1.5.1
**Updated from an older ownCloud or fresh install:**
updated stepwise from 9.1.0
**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

**CardDAV-clients:**

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by DeepDiver1975** [comment]
mind testing this version? [contacts.tar.gz.txt](https://github.com/owncloud/contacts/files/1107022/contacts.tar.gz.txt)

**Comment by DeepDiver1975** [comment]
should be fixed with 1.5.3 -> closing
